caskadht.go

```diff
@@ -182,6 +184,16 @@ LOOP:
 				}
 				provider.Addrs = found.Addrs
 			}
+
+			if !c.addrFilterDisabled {
+				provider.Addrs = multiaddr.FilterAddrs(provider.Addrs, manet.IsPublicAddr)
```

Comment:
This will remove valid DNS addresses. What is really needed here is something like this:
```go
notPrivateAddr := func(a multiaddr.Multiaddr) bool { return !manet.IsPrivateAddr(a) }
provider.Addrs = multiaddr.FilterAddrs(provider.Addrs, notPrivateAddr)
```

Reply:
You are right; thank you

Reply:
Unfortunately private Addr check doesnt check for unroutable ones...

So what we need is: "is public" or is DNS addr I _think_.